accounts-db/src/accounts.rs

```diff
@@ -514,9 +514,9 @@ impl Accounts {
     /// This function will prevent multiple threads from modifying the same account state at the
     /// same time
     #[must_use]
-    pub fn lock_accounts<'a>(
+    pub fn lock_accounts<'a, Tx: SVMMessage + 'a>(
```

Comment:
`SVMMessage` is not a type, it's a trait - whatever it is, it could technically have any lifetime attached to it since the type is abstract and I put no lifetime bound on the trait itself.
So we need **some** bound that it lives at least as long as the reference itself (which I don't see how it could not?), but the compile fails to do it automatically for us.

This is only necessary when it's in an iterator (or probably some other trait), because the compiler is unable to infer the lifetime like it is when we pass an single `&impl SVMMessage`.